deltachat-ios/AppDelegate.swift

```diff
@@ -445,24 +450,54 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
 
     // MARK: - misc.
 
-    func openDatabase() {
-        guard let databaseLocation = DatabaseHelper(dcContext: dcContext).updateDatabaseLocation() else {
+    func migrateToDcAccounts() {
+        let dcContext = DcContext()
+
+        // first database migration to shared container, can be removed at some point
+        // implemented in April 2020 (https://github.com/deltachat/deltachat-ios/pull/612)
+        let databaseHelper = DatabaseHelper(dcLogger: dcContext.logger)
+        guard let databaseLocation = databaseHelper.updateDatabaseLocation() else {
             fatalError("Database could not be opened")
         }
-        logger.info("open: \(databaseLocation)")
         dcContext.openDatabase(dbFile: databaseLocation)
+        if dcContext.isConfigured() {
```

Comment:
i think, it is not needed to try to open the database and check for configured, esp. as this would also happen when the file is no longer existant. i would just check for existence of the old database file and pass it to migrateAccount() then.